__e2e__/init.test.ts

```diff
@@ -39,6 +39,15 @@ afterEach(() => {
   cleanupSync(DIR);
 });
 
+test('init fails if the directory already exists', () => {
+  fs.mkdirSync(path.join(DIR, 'TestInit'));
+
+  const {stderr} = runCLI(DIR, ['init', 'TestInit'], {expectedFailure: true});
```

Comment:
Let's use a custom local template, so we don't pay the cost of fetching all of those dependencies all over again for another test. See: https://github.com/react-native-community/cli/pull/1329/files#diff-f0cd47ce844b386b0a1e17b5f44a631eaf7bdc98c5b0e4d3df7dd674e92ed87dR67-R72

Reply:
What it does is creating an empty directory, and making sure, an attempt to create a project under the same name errors. Since this is the case of failure, I don't think supplying the `--template` flag is necessary.

Reply:
Ugh, brainfart